### PR TITLE
Add keyword analysis to edn

### LINKED ISSUE #1926: Keyword analysis for EDN
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! You can remove this message. ]

**Is your feature request related to a problem? Please describe.**

Implementing this allows it to be possible for clojure-lsp to provide things like outlines and references for EDN files.

**Describe the solution you'd like**

I'm not sure, I'm not familiar with kondo's implementation :sweat_smile: 

**Describe alternatives you've considered**

I started writing my own minor mode to try and handle this with regexp, but it seemed more appropriate to have this implemented upstream

**Additional context**

There seems to be some overlap with https://github.com/clj-kondo/clj-kondo/issues/1902, which could bridge that gap

I am fine to try and look into adding this myself, if necessary.

### PR DESCRIPTION
**Pull request by ericdallo**
Please answer the following questions and leave the below in as part of your PR.

- [x] I have read the [developer documentation](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md).

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

- [x] This PR contains a [test](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#tests) to prevent against future regressions

- [X] I have updated the [CHANGELOG.md](https://github.com/clj-kondo/clj-kondo/blob/master/CHANGELOG.md) file with a description of the addressed issue.

Fixes #1926 

I did some tests with clojure-lsp and this is enough to make find-references of keywords in any edn file :)

### CODE REVIEW

File: src/clj_kondo/impl/analyzer.clj [hunk 1]
```diff
@@ -2597,15 +2597,14 @@
...
-          (when-not (= :edn (:lang ctx))
-            (let [id (gensym)
-                  expr (assoc expr :id id)
-                  _ (analyze-usages2 ctx expr)
-                  usage (get @(:calls-by-id ctx) id)]
-              (if usage
-                (types/add-arg-type-from-usage ctx usage expr)
-                (types/add-arg-type-from-expr ctx expr))
-              nil)))
+          (let [id (gensym)
```
Comment (borkdude) [inline src/clj_kondo/impl/analyzer.clj:2600]: Maybe we can keep the old behavior when keyword analysis isn't enabled since this will be more performant

Comment (ericdallo) [reply src/clj_kondo/impl/analyzer.clj:2600]: Makes sense, a little bit odd make a check if keywords is enabled here though

Comment (borkdude) [reply src/clj_kondo/impl/analyzer.clj:2600]: Why?

Comment (ericdallo) [reply src/clj_kondo/impl/analyzer.clj:2600]: Not a big deal, it's just that seems a little bit coupled.
I can do that

Comment (borkdude) [reply src/clj_kondo/impl/analyzer.clj:2600]: The goal is to not do extra work when you're not using keyword analysis. How this is accomplished is secondary to me.

**Comment by ericdallo** [comment]
Is this good to merge @borkdude ?

**Comment by borkdude** [comment]
I think it looks good but I want to run some more tests here, will do this tomorrow

**Comment by borkdude** [comment]
Made some minor tweaks and merged to master

**Pull request closed without merging**